main/main.c

```diff
@@ -186,5 +259,9 @@ main(int argc, char **argv)
 net:
 	gatekeeper_free_network();
 out:
+	rte_free(lua_base_dir);
+	lua_base_dir = NULL;
+	rte_free(gatekeeper_config_file);
+	gatekeeper_config_file = NULL;
```

Comment:
Once the global variables are initialized differently (see another comment), we no longer need to deallocate memory here.